src/openpgp.js

```diff
@@ -240,24 +240,26 @@ export async function encryptKey({ privateKey, passphrase }) {
- * @param  {Signature} signature                      (optional) a detached signature to add to the encrypted message
- * @param  {Boolean} wildcard                         (optional) use a key ID of 0 instead of the public key IDs
- * @param  {Date} date                                (optional) override the creation date of the message signature
- * @param  {Array<Object>} fromUserIds                (optional) array of user IDs to sign with, one per key in `privateKeys`, e.g. [{ name:'Steve Sender', email:'steve@openpgp.org' }]
- * @param  {Array<Object>} toUserIds                  (optional) array of user IDs to encrypt for, one per key in `publicKeys`, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
+ * @param  {Message} message                            message to be encrypted as created by openpgp.Message.fromText or openpgp.Message.fromBinary
+ * @param  {Key|Array<Key>} publicKeys                  (optional) array of keys or single key, used to encrypt the message
+ * @param  {Key|Array<Key>} privateKeys                 (optional) private keys for signing. If omitted message will not be signed
+ * @param  {String|Array<String>} passwords             (optional) array of passwords or a single password to encrypt the message
+ * @param  {Object} sessionKey                          (optional) session key in the form: { data:Uint8Array, algorithm:String }
+ * @param  {module:enums.compression} compression       (optional) which compression algorithm to compress the message with, defaults to what is specified in config
+ * @param  {Boolean} armor                              (optional) whether the return values should be ascii armored (true, the default) or binary (false)
+ * @param  {'web'|'ponyfill'|'node'|false} streaming    (optional) whether to return data as a stream. Defaults to the type of stream `message` was created from, if any.
+ * @param  {Signature} signature                        (optional) a detached signature to add to the encrypted message
+ * @param  {Boolean} wildcard                           (optional) use a key ID of 0 instead of the public key IDs
+ * @param  {Date} date                                  (optional) override the creation date of the message signature
+ * @param  {Array<Object>} fromUserIds                  (optional) array of user IDs to sign with, one per key in `privateKeys`, e.g. [{ name:'Steve Sender', email:'steve@openpgp.org' }]
+ * @param  {Array<Object>} toUserIds                    (optional) array of user IDs to encrypt for, one per key in `publicKeys`, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
+ * @param  {Array<module:type/keyid>} encryptionKeyIds  (optional) array of key IDs to use for encryption. Each encryptionKeyIds[i] corresponds to publicKeys[i]
+ * @param  {Array<module:type/keyid>} signingKeyIds     (optional) array of key IDs to use for signing. Each signingKeyIds[i] corresponds to privateKeys[i]
```

Comment:
And then also here, I would order them `signingKeyIds`, `encryptionKeyIds`, `date`, `fromUserIds`, `toUserIds`, even though it looks a bit weird, to be consistent with e.g. `getEncryptionKey`. (And idem below.)

Reply:
I reordered the parameters in `openpgp.encrypt` and `Message.encrypt`. Reordering the parameters in a similar fashion in `Message.sign` or `Message.encryptSessionKey` would result in failing unit tests. Refactoring must be expensive, thoughts 🤔?

Reply:
Oh, yeah, I guess there are some tests that call `message.sign` etc directly, maybe? I think it's still fine to update these functions, and just pass `undefined` in the correct spot in the tests to make them pass. These functions are considered part of the low-level API, so it's not a gigantic deal to break them, and v5 will be a breaking release, anyway 😌

Internally within OpenPGP.js, I think there are only a few places that call `Message.sign` or `Message.encryptSessionKey`, so I think it should be manageable, but lmk if I missed something.

Reply:
Sounds simple enough :)

Reply:
Sorry for the delay! I reordered the parameters in `message.sign`, `message.signDetached`, `createEncryptionPackets` and `openpgp.sign`

Reply:
Thanks a lot!